extras/drift_postgres/lib/src/pg_database.dart

```diff
@@ -0,0 +1,202 @@
+import 'package:drift/backends.dart';
+import 'package:postgres/postgres.dart';
+
+///
+class PgDatabase extends DelegatedDatabase {
+  ///
+  PgDatabase(PostgreSQLConnection connection)
+      : super(_PgDelegate(connection, connection),
+            isSequential: true, logStatements: true);
+
+  ///
+  factory PgDatabase.open(String host, int port, String databaseName,
+      {String? username,
+      String? password,
+      int timeoutInSeconds = 30,
+      int queryTimeoutInSeconds = 30,
+      String timeZone = 'UTC',
+      bool useSSL = false,
+      bool isUnixSocket = false}) {
+    return PgDatabase(PostgreSQLConnection(host, port, databaseName,
+        username: username,
+        password: password,
+        timeoutInSeconds: timeoutInSeconds,
+        queryTimeoutInSeconds: queryTimeoutInSeconds,
+        timeZone: timeZone,
+        useSSL: useSSL,
+        isUnixSocket: isUnixSocket));
+  }
+}
+
+///
+class _PgDelegate extends DatabaseDelegate {
+  final PostgreSQLConnection _db;
+  final PostgreSQLExecutionContext _ec;
+
+  @override
+  SqlDialect get dialect => SqlDialect.postgres;
+
+  _PgDelegate(this._db, this._ec) : closeUnderlyingWhenClosed = false;
+
+  bool _isOpen = false;
+
+  final bool closeUnderlyingWhenClosed;
+
+  @override
+  TransactionDelegate get transactionDelegate => _PgTransactionDelegate(_db);
+
+  @override
+  late DbVersionDelegate versionDelegate;
+
+  @override
+  Future<bool> get isOpen => Future.value(_isOpen);
+
+  @override
+  Future<void> open(QueryExecutorUser user) async {
+    final pgVersionDelegate = _PgVersionDelegate(_db);
+
+    await _db.open();
+    await pgVersionDelegate.init();
+    await _initializeDatabase();
+
+    versionDelegate = pgVersionDelegate;
+    _isOpen = true;
+  }
+
+  Future _ensureOpen() async {
+    if (_db.isClosed) {
+      await _db.open();
+    }
+  }
+
+  Future<void> _initializeDatabase() async {
+    // TODO: Do we need create these functions?
+    //setup?.call(_db);
+  }
+
+  final _regexIndexed = RegExp(r'\?(\d+)');
+  final _regexNamed = RegExp(r':(\w+)');
+
+  String _convertStatement(String statement) {
+    final argMap = {};
+    return statement
+        .replaceAllMapped(_regexIndexed, (match) => '@${match[1]}')
+        .replaceAllMapped(_regexNamed, (match) {
+      final index = argMap.putIfAbsent(match[1], () => argMap.length + 1);
+      return '@$index';
+    });
+  }
```

Comment:
I don't like that we have to do this mapping, it would break if someone writes something like `':sunglasses:'` in a string literal. We can change how variables are written at runtime, right? Is this as a guard for compiled queries?

Reply:
It is a way to not to rewrite all queries and related code in Drift and keep backward compatibility. SQLite understand @ and ? too, but Postgres don't accept '?'. Also Postgres not understands named parameters. I rewrited the related code before, but reverted and I made this replace script. Would be a big change to use '@' and remove named parameters. `:sunglasses:` isn't acceptable by SQLite either (https://www.sqlite.org/c3ref/bind_blob.html). Only alphanumeric characters accepted. `:` at the end throw error. Also, the regex not match it, so won't be replaced to `@sunglasses@`. I think this script is the best we have now. Maintains named parameters' order and keeps backward compatibility for current code.

Reply:
Sorry, I was being unclear. My point is that the following query does not contain any variables: `SELECT ':sunglasses:'` (it returns a string with the content `:sunglasses:`). After your transformation, we would run `SELECT '?'` which returns a different value!

We can change the `Variable` class to write postgres-compatible parameters. Generated code will be a challenge, of course - I'll take a look and see what can be done about this.

Reply:
Yes you sure. I didn't thought about it. I already made a full rewrite from '?' to '@' and it's worked like a charm with SQLite and Postgres too. But I reverted for backward compatibility. Can we make a major version change and migrate users to @ (or $)? As I see, SQL parsed in moor files and in Driftdatabase annotation. There It can be automatically replace `?` to `@` in generated code. This change would affect only custom queries.